crates/broker/src/order_picker.rs

```diff
@@ -90,13 +91,20 @@ where
             }
         }
 
-        // TODO(#BM-536): Filter based on supported selectors
-        // Drop orders that specify a selector
-        if order.request.requirements.selector != FixedBytes::<4>([0; 4]) {
-            tracing::warn!("Removing order {order_id:x} because it has a selector requirement");
-            self.db.skip_order(order_id).await.context("Order has a selector requirement")?;
-            return Ok(());
-        }
+        match Selector::from_bytes(order.request.requirements.selector.into())
+            .context("Failed to parse selector")?
+        {
+            // Supported selectors
+            Selector::FakeReceipt | Selector::SetVerifierV0_2 | Selector::Groth16V1_2 => {}
```

Comment:
`Selector::FakeReceipt` is all zeroes, which is a bit confusing in this context being it will interpret requests with no particular selector as a request for a "fake receipt". Splitting off into its own case would be more clear.
```suggestion
            // Default: any selector is valid
            [0u8; 4] => {}
            Selector::SetVerifierV0_2 | Selector::Groth16V1_2 => {}
```
(I don't think that suggestion will actually compile)

Reply:
I've created a `UNSPECIFIED_SELECTOR` constant